File: chat_rn/src/models/chatMessage.ts
export type MessageRole = 'user' | 'assistant';

export interface ChatMessage {
  id: string;
  role: MessageRole;
  text: string;
  timestamp: Date;
  isError?: boolean;
}

export function isUserMessage(message: ChatMessage): boolean {
  return message.role === 'user';
}

File: chat_app/lib/models/chat_message.dart
enum MessageRole { user, assistant }

class ChatMessage {
  const ChatMessage({
    required this.id,
    required this.role,
    required this.text,
    required this.timestamp,
    this.isError = false,
  });

  final String id;
  final MessageRole role;
  final String text;
  final DateTime timestamp;
  final bool isError;

  bool get isUser => role == MessageRole.user;
}
